File: need4spd/euler_38.py
def isPandigital(n):
  if (len(n) == 9):
    temp_l = ''.join(sorted([ c for c in n]))
    if temp_l == '123456789':
      return True;
  return False
  

result = 0
for n in range(9183, 9877):
  a1 = str(n * 1) + str(n * 2)
  if isPandigital(a1):
    if result < int(a1):
      result = int(a1)

print (result)

File: need4spd/euler_38.pl
sub isPandigital {
    my ($n) = @_;
    if (length $n == 9) {
        my @temp_l = split //, $n;
        @temp_l = sort @temp_l;
        
        my $temp_s = join("", @temp_l);
        if ($temp_s =~ m/123456789/) {
            return 1;
        }
    }
    
    return 0;
}

my $result = 0;

foreach my $n ((9183..9877)) {
    my $a1 = ($n*1).($n*2);
    
    if (isPandigital($a1)) {
        if ($result < $a1) {
            $result = $a1;
        }
    }
}

print "$result";